snippets/snippets.json

```diff
@@ -117,115 +107,795 @@
-        "prefix": "literal",
+	},
+	"Insert literal block": {
+		"prefix": "literal",
 		"body": [
-            "[${1:attributes}]",
+			"[${1:attributes}]",
 			"....",
 			"${2:${TM_SELECTED_TEXT:text}}",
 			"....",
 			"$0"
 		],
 		"description": "Insert literal block"
-    },
-    "Insert source block": {
-        "prefix": "source",
+	},
+	"Insert source block": {
+		"prefix": "source",
 		"body": [
-            "[source,${1:language},${2:attributes}]",
+			"[source,${1:language},${2:attributes}]",
 			"----",
 			"${3:${TM_SELECTED_TEXT:code}}",
 			"----",
 			"$0"
 		],
 		"description": "Insert source block"
-    },
-    "Insert example block": {
-        "prefix": "example",
+	},
+	"Insert example block": {
+		"prefix": "example",
 		"body": [
-            "[${1:attributes}]",
+			"[${1:attributes}]",
 			"====",
 			"${2:${TM_SELECTED_TEXT:text}}",
 			"====",
 			"$0"
 		],
 		"description": "Insert example block"
-    },
-    "Insert sidebar block": {
-        "prefix": "sidebar",
+	},
+	"Insert sidebar block": {
+		"prefix": "sidebar",
 		"body": [
-            "[${1:attributes}]",
+			"[${1:attributes}]",
 			"****",
 			"${2:${TM_SELECTED_TEXT:text}}",
 			"****",
 			"$0"
 		],
 		"description": "Insert sidebar block"
-    },
-    "Insert table": {
-        "prefix": "table",
-        "body": [
-            "[${1:Attributes}]",
-            "|===",
-            "|${2:Header column 1} |${3:Header column 2} |${4:Header column 3}",
-            "",
-            "|${5:Cell 1, row 1}",
-            "|${6:Cell 2, row 1}",
-            "|${7:Cell 3, row 1}",
-            "|===",
-            "$0"
-        ],
-        "description": "Insert table"
-    },
-    "Insert toc": {
-        "prefix": ":toc:",
-        "body": ":toc: ${1|auto,preamble,macro,right,left|}\n",
-        "description": "Insert toc attribute"
-    },
-    "Insert doctype": {
-        "prefix": ":doctype:",
-        "body": ":doctype: ${1|article,book,manpage,inline|}\n",
-        "description": "Insert doctype attribute"
-    },
-    "Insert source highlighter": {
-        "prefix": ":source-highlighter:",
-        "body": ":source-highlighter: ${1|coderay,highlightjs,prettify,pygments,rouge|}\n",
-        "description": "Insert source highlighter"
-    },
-    "Insert attribute": {
-        "prefix": ":attribute:",
-        "body": ":${1:attribute}: ${2:value}",
-        "description": "Insert attribute"
-    },
-    "Insert tagged block": {
-        "prefix": "tag",
+	},
+	"Insert table": {
+		"prefix": "table",
 		"body": [
-            "// tag::${1:tagname}[]",
-            "${2:${TM_SELECTED_TEXT:text}}",
-            "// end::${1:tagname}[]",
+			"[${1:Attributes}]",
+			"|===",
+			"|${2:Header column 1} |${3:Header column 2} |${4:Header column 3}",
+			"",
+			"|${5:Cell 1, row 1}",
+			"|${6:Cell 2, row 1}",
+			"|${7:Cell 3, row 1}",
+			"|===",
+			"$0"
+		],
+		"description": "Insert table"
+	},
+	"Insert source highlighter": {
```

Comment:
Maybe we should use the verb "Declare" or "Define" when we declare/define an attribute?

Reply:
Yeah, there's probably something better than just "insert" everywhere. I think it would make sense to change the wording based on the type of the attribute, e.g., "Enable :hide-uri-scheme:" for a bool vs "choose :docinfo:" for an enum vs "set :author:" for an arbitrary string.

Reply:
Yes that's a good idea but let's keep it for now, we can improve it later.